### ISSUE TITLE
Angular 2?

### ISSUE THREAD

**Issue by jonathandick** [kentcdodds/ama#66]
Kent, 

We're beginning the planning phase for shifting some work to angular 2 now that the beta version is out. Any thoughts on how ready it is to use for production apps?

Are you likely to create a formly version in angular 2?

Thanks,

Jonathan

**Comment by kentcdodds** [comment]
Hi @jonathandick. Thanks for the question.

So I actually talked about my feelings about Angular 2 a bit in [this question](https://github.com/kentcdodds/ama/issues/7). Basically, Angular 2 is amazing. There's no denying that. And I know that there are some things we don't know about Angular 2 yet that are going to be even more incredible. The challenge is time. I just don't have time to do everything under the sun I'd like to do (see [FOMO](https://github.com/kentcdodds/ama/issues/51)). So I've decided to go pretty much all React. We'll see what the future holds for me.

On angular-formly, I actually had a question about this specifically (#22). But my answer has changed a bit now. The fact of the matter is I only have so much [Open Source Stamina](http://kcd.im/stamina). I don't think that I'll be using angular-formly again and I don't know what the future holds for my work with Angular 2. If I end up using Angular 2, then I may work on angular2-formly. Otherwise I doubt that I'll be the right person for the job (see that post).

To answer your first question more directly... Whether Angular 2 is ready for production is (IMO) answered by the fact that it's still in `beta`. Not quite ready, but really close. If you have something that doesn't need to be out in 6 months, I'd say go for it if that's what you want to use!

Good luck!

**Comment by rayhooker** [comment]
Kent,  React is something I have considered.  Any pointers to options that are similar or address the same need for data driven forms for schema and composition? I see react-json-editor (https://github.com/ismaelga/react-json-editor).

**Comment by kentcdodds** [comment]
There actually is a [react-formly](https://github.com/formly-js/react-formly) which I made, but I wouldn't recommend it. It's too immature and I don't think anyone's really using it. I saw a local talk about [react-loose-forms](https://github.com/espeakers/react-loose-forms) and that looked interesting, but I haven't used it.

Unfortunately I haven't really had a use case for forms in React yet, so I can't give you any good recommendations. Good luck!
